prime-router/src/test/kotlin/azure/SettingFacadeTests.kt

```diff
@@ -193,6 +193,14 @@ class SettingFacadeTests {
         assertThat(sender).isNull()
     }
 
+    @Test
+    fun `get sender test failure with invalid name`() {
+        setupSenderDatabaseAccess()
+        // verifies solve for #4002 - an overly-long exception name should not be throwing an exception, instead failing as normal
+        val sender = SettingsFacade(testMetadata(), accessSpy).findSender("ignore.ignore-waters.ignore-waters")
```

Comment:
Incomplete unit test.  You need to test both good and bad inputs.  What happens if the service is not specified, but the org is (e.g. "ignore" only)?